### ISSUE TITLE
Commit Staged (Amend) doesn't change the commit message if there are no staged changes but there are changes

### ISSUE THREAD

**Issue by Maxpxt** [microsoft/vscode#174195]
Does this issue occur when all extensions are disabled?: Yes/No

- VS Code Version: 1.75.1 (user setup)
- OS Version: Windows_NT x64 10.0.22621

Steps to Reproduce:

1. Open vscode on a git repository.
2. Do some changes, but don't stage anything.
3. In the SCM view, write a commit message in the input field then select "Commit > Commit Staged All (Amend)" in the context menu from the "..." icon.

The expected behavior is for the message of the last commit to be updated.

What actually happens is nothing. The message of the last commit is not updated and no command is shown in the git output tab.

Note that when there are staged changes or when there are no changes at all, staged and not staged, "Commit > Commit Staged (Amend)" does update the message as expected. It only does nothing when there are changes, but no staged changes.

This bug was previously tracked in #120639. That issue was closed and locked, but the bug was not fully solved. "Commit > Commit All (Amend)" now works, but "Commit > Commit *Staged* All (Amend)" still does not work.

Does this issue occur when all extensions are disabled?: Yes

**Comment by Maxpxt** [comment]
@lszomoru (pinging because you were taking care of #120693)

**Comment by joaomoreno** [comment]
I don't think we have a `Commit Staged All (Amend)` command... only `Commit All (Amend)` and `Commit Staged (Amend)`. Which one do you mean @Maxpxt?

**Comment by Maxpxt** [comment]
@joaomoreno Yes, that was a typo, sorry. I mean `Commit Staged (Amend)`

@lszomoru 

> My understanding is that all commands that have "Staged" in their name operate on the resources that are in the "Staged Changes" resource group. Since in the scenario above there are no staged resources, the command bails.

Yes, however, `Commit Staged (Amend)` currently does not bail if there are no changes at all (none staged and none in the working tree), and properly amends the commit message.

Also, with the current behavior, if there are changes in the working tree, the only way of amending the commit message is to discard or stash those changes.
